include/swift/Runtime/Win32Strings.h

```diff
@@ -0,0 +1,35 @@
+//===--- Win32Strings.h ---------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// Includes conversion functions for use with Windows API that produces or
+// consumes wide strings (wchar_t) which, on Windows, are UTF-16.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_RUNTIME_WIN32STRINGS_H
+#define SWIFT_RUNTIME_WIN32STRINGS_H
+
+#if defined(_WIN32)
+namespace swift {
+/// Convert an argument, represented by a wide string, to UTF-8.
+///
+/// @param str The string to convert.
+///
+/// @returns The string, converted to UTF-8. The caller is responsible for
+///   freeing this string when done with it.
+///
+/// If @a str cannot be converted to UTF-8, @c nullptr is returned.
+SWIFT_RUNTIME_STDLIB_INTERNAL
+char *swift_copyUTF8FromWide(const wchar_t *str);
+}
+#endif
+#endif
```

Comment:
Can we get away with something simpler?  Namely using LLVMSupport.  That has a nice helper `ConvertUTF16ToUTF8`.

Reply:
I'm definitely open to reusing some other logic here instead of relying on Win32 API.

That function's in LLVM-proper, I believe, so not available at runtime in Swift. Are you saying it _can_ be used from the runtime?

Reply:
We do have a copy of LLVMSupport that we have slightly modified (to avoid ODR violations) that we use in the runtime.  We did some amount of DCE so I'm not sure if the function is readily available, but in theory it can be.

Reply:
Yeah, we don't want to be using `operator new` if we can help it.

But also, we shouldn't be duplicating this work — I've got a version of the same thing in https://github.com/apple/swift/pull/62462. Perhaps we could pull the character conversion code out into a separate PR?

Reply:
To add to that, _personally_ I'm not keen on copying _more_ things from LLVM into LLVMSupport. They aren't kept in sync with LLVM proper (unless someone checks and updates them) and they make it look like we can use LLVM things in the runtime, which we generally can't.